aws-datastore/src/test/java/com/amplifyframework/datastore/AWSDataStorePluginTest.java

```diff
@@ -52,7 +61,6 @@
 @RunWith(RobolectricTestRunner.class)
 public final class AWSDataStorePluginTest {
```

Comment:
I think we should start splitting tests out into other classes. The ones I have in here right now are related to model synchronization. Probably `AWSDataStorePluginTest` was a bad name to begin with. All in all, the plugin does a lot of different stuff. I think we are ultimately funneling all of our high level tests into this class and its going to end up having a lot of unrelated stuff and being a bit of a maintenance issue.

Reply:
If it's ok, I'd like to undertake that as a separate effort. At the very least, we could break it up in API-mode tests vs non-API-mode tests, but it can be broken down much further.

Reply:
Sounds good to me